amm/contracts/router/lib.rs

```diff
@@ -67,21 +75,48 @@ pub mod router {
             Ok(())
         }
 
+        #[ink(message)]
+        pub fn add_stable_to_cache(&mut self, pool: AccountId) -> Result<(), CallerIsNotOwner> {
+            ensure!(self.env().caller() == self.owner, CallerIsNotOwner);
+            self.cache_stable(pool);
+            Ok(())
+        }
+
         #[ink(message)]
         pub fn remove_pair_from_cache(
             &mut self,
             token_0: AccountId,
             token_1: AccountId,
         ) -> Result<(), CallerIsNotOwner> {
             ensure!(self.env().caller() == self.owner, CallerIsNotOwner);
-            self.pairs.remove((token_0, token_1));
-            self.pairs.remove((token_1, token_0));
+            if let Some((pair, _)) = self.cached_pairs.take((token_0, token_1)) {
+                self.cached_pools.remove(pair);
+            }
+            self.cached_pairs.remove((token_1, token_0));
```

Comment:
We need to remove `(token_0, token_1)` as well.

Reply:
I think `self.cached_pairs.take((token_0, token_1))` in line 92 already removes `(token_0, token_1)` :thinking:

Reply:
It does.